### DISCUSSION TITLE
x-on:click does nothing

### DISCUSSION THREAD

**Discussion by blainsmith** [alpinejs/alpine#2777] [7. Bugs (archived)] [answered]
```html
<html>
    <head>
        <title>Click</title>

        <script src="https://unpkg.com/alpinejs" defer></script>
    </head>
    <body>
        <button x-on:click="alert('hi')">Hi</button>
    </body>
</html>
```

No matter where I use `x-on:click` or `@click` nothing happens. It doesn't even look like an event listeners get attached to the elements.

I am using the latest Chrome release.

**Comment by browner12** [comment, marked as answer]
you need to add an `x-data` attribute to a surrounding element.

**Reply by blainsmith** [reply]
Ah interesting ok, thanks! Did I miss that in the docs somewhere or is it assumed everything should be surrounded by an element with `x-data`?

**Reply by browner12** [reply]
https://alpinejs.dev/directives/data

**Reply by blainsmith** [reply]
"Everything in Alpine starts with the x-data directive."

🤦 

Thank you.
